framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -195,9 +196,37 @@ private void viewpointAdaptExecutable(
             receiver = FlowExpressions.internalReprOf(factory, receiverTree);
         }
 
-        List<FlowExpressions.Receiver> argReceivers = new ArrayList<>(args.size());
-        for (ExpressionTree argTree : args) {
-            argReceivers.add(FlowExpressions.internalReprOf(factory, argTree));
+        List<FlowExpressions.Receiver> argReceivers = new ArrayList<>();
+        boolean varargs = false;
+        if (tree instanceof MethodInvocationTree) {
+            ExecutableElement methodCalled =
+                    ((ExecutableElement)
+                            TreeUtils.elementFromUse(
+                                    ((MethodInvocationTree) tree).getMethodSelect()));
+            if (methodCalled.isVarArgs()) {
+                varargs = true;
+                for (int i = 0; i < methodCalled.getParameters().size() - 1; i++) {
+                    argReceivers.add(FlowExpressions.internalReprOf(factory, args.get(i)));
+                }
+                List<Receiver> initializers = new ArrayList<>();
+                for (int i = methodCalled.getParameters().size() - 1; i < args.size(); i++) {
+                    initializers.add(FlowExpressions.internalReprOf(factory, args.get(i)));
+                }
+                TypeMirror tm =
+                        ElementUtils.getType(
+                                methodCalled
+                                        .getParameters()
+                                        .get(methodCalled.getParameters().size() - 1));
```

Comment:
Can you split this into one or more local variables so that the line isn't broken up like this? As-is it's completely unreadable.